### DISCUSSION TITLE
Compilation warnings with Xcode 16.0 build 16A242d

### DISCUSSION THREAD

**Discussion by wyatt-wong** [mamedev/discussions#151] [General]
I compile MAME in macOS using Xcode 16.0 build 16A242d and saw a lot of warning messages. Would someone please change those deprecated code so that they are no long marked as deprecated in latest version of Xcode ?

Compiling src/mame/pc/calchase.cpp...
../../../../../src/mame/pc/calchase.cpp:410:5: warning: 'pcat_base_state' is deprecated: Leaky abstraction of a southbridge, to be replaced with actual chipset emulation. [-Wdeprecated-declarations]
  410 |                 : pcat_base_state(mconfig, type, tag)
      |                   ^
In file included from ../../../../../src/mame/pc/calchase.cpp:125:
../../../../../src/mame/shared/pcshare.h:19:4: note: 'pcat_base_state' has been explicitly marked deprecated here
   19 |         [[deprecated("Leaky abstraction of a southbridge, to be replaced with actual chipset emulation.")]]
      |           ^

Compiling src/mame/misc/a1supply.cpp...
Archiving libmitsubishi.a...
../../../../../src/mame/pc/fruitpc.cpp:40:5: warning: 'pcat_base_state' is deprecated: Leaky abstraction of a southbridge, to be replaced with actual chipset emulation. [-Wdeprecated-declarations]
   40 |                 : pcat_base_state(mconfig, type, tag)
      |                   ^
In file included from ../../../../../src/mame/pc/fruitpc.cpp:23:
../../../../../src/mame/shared/pcshare.h:19:4: note: 'pcat_base_state' has been explicitly marked deprecated here
   19 |         [[deprecated("Leaky abstraction of a southbridge, to be replaced with actual chipset emulation.")]]
      |           ^

Compiling src/mame/technos/wwfsstar.cpp...
../../../../../src/mame/pc/pangofun.cpp:115:5: warning: 'pcat_base_state' is deprecated: Leaky abstraction of a southbridge, to be replaced with actual chipset emulation. [-Wdeprecated-declarations]
  115 |                 : pcat_base_state(mconfig, type, tag) { }
      |                   ^
In file included from ../../../../../src/mame/pc/pangofun.cpp:102:
../../../../../src/mame/shared/pcshare.h:19:4: note: 'pcat_base_state' has been explicitly marked deprecated here
   19 |         [[deprecated("Leaky abstraction of a southbridge, to be replaced with actual chipset emulation.")]]
      |           ^

Compiling src/mame/pacman/pacman.cpp...
../../../../../src/mame/pc/paokaipc.cpp:38:5: warning: 'pcat_base_state' is deprecated: Leaky abstraction of a southbridge, to be replaced with actual chipset emulation. [-Wdeprecated-declarations]
   38 |                 : pcat_base_state(mconfig, type, tag)
      |                   ^
In file included from ../../../../../src/mame/pc/paokaipc.cpp:23:
../../../../../src/mame/shared/pcshare.h:19:4: note: 'pcat_base_state' has been explicitly marked deprecated here
   19 |         [[deprecated("Leaky abstraction of a southbridge, to be replaced with actual chipset emulation.")]]
      |           ^

Compiling src/mame/pc/pcat_nit.cpp...
../../../../../src/mame/pc/pcat_dyn.cpp:49:5: warning: 'pcat_base_state' is deprecated: Leaky abstraction of a southbridge, to be replaced with actual chipset emulation. [-Wdeprecated-declarations]
   49 |                 : pcat_base_state(mconfig, type, tag)
      |                   ^
In file included from ../../../../../src/mame/pc/pcat_dyn.cpp:25:
../../../../../src/mame/shared/pcshare.h:19:4: note: 'pcat_base_state' has been explicitly marked deprecated here
   19 |         [[deprecated("Leaky abstraction of a southbridge, to be replaced with actual chipset emulation.")]]
      |           ^
../../../../../src/mame/pc/pcat_nit.cpp:102:5: warning: 'pcat_base_state' is deprecated: Leaky abstraction of a southbridge, to be replaced with actual chipset emulation. [-Wdeprecated-declarations]
  102 |                 : pcat_base_state(mconfig, type, tag),
      |                   ^
In file included from ../../../../../src/mame/pc/pcat_nit.cpp:88:
../../../../../src/mame/shared/pcshare.h:19:4: note: 'pcat_base_state' has been explicitly marked deprecated here
   19 |         [[deprecated("Leaky abstraction of a southbridge, to be replaced with actual chipset emulation.")]]
      |           ^

Compiling src/mame/sun/sun2.cpp...
../../../../../src/mame/pc/queen.cpp:48:5: warning: 'pcat_base_state' is deprecated: Leaky abstraction of a southbridge, to be replaced with actual chipset emulation. [-Wdeprecated-declarations]
   48 |                 : pcat_base_state(mconfig, type, tag)
      |                   ^
In file included from ../../../../../src/mame/pc/queen.cpp:34:
../../../../../src/mame/shared/pcshare.h:19:4: note: 'pcat_base_state' has been explicitly marked deprecated here
   19 |         [[deprecated("Leaky abstraction of a southbridge, to be replaced with actual chipset emulation.")]]
      |           ^

Compiling src/mame/televideo/ts3000.cpp...
../../../../../src/mame/misc/blitz68k.cpp:1610:81: warning: 'field_set_analog_reset' is deprecated: PORT_RESET is deprecated; manage counter state explicitly [-Wdeprecated-declarations]
 1610 |         PORT_BIT( 0x0f, 0x00, IPT_TRACKBALL_X ) PORT_SENSITIVITY(2) PORT_KEYDELTA(100) PORT_RESET
      |                                                                                        ^
In file included from ../../../../../src/mame/misc/blitz68k.cpp:1:
In file included from /Users/wyattwong/GitHub/mame/build/projects/sdl/mame/gmake-osx-clang/../../../../../src/emu/emu.h:58:
/Users/wyattwong/GitHub/mame/build/projects/sdl/mame/gmake-osx-clang/../../../../../src/emu/ioport.h:1202:13: note: expanded from macro 'PORT_RESET'
 1202 |         configurer.field_set_analog_reset();
      |                    ^
/Users/wyattwong/GitHub/mame/build/projects/sdl/mame/gmake-osx-clang/../../../../../src/emu/ioport.h:1059:4: note: 'field_set_analog_reset' has been explicitly marked deprecated here
 1059 |         [[deprecated("PORT_RESET is deprecated; manage counter state explicitly")]]
      |           ^
../../../../../src/mame/misc/blitz68k.cpp:1613:81: warning: 'field_set_analog_reset' is deprecated: PORT_RESET is deprecated; manage counter state explicitly [-Wdeprecated-declarations]
 1613 |         PORT_BIT( 0x0f, 0x00, IPT_TRACKBALL_Y ) PORT_SENSITIVITY(2) PORT_KEYDELTA(100) PORT_RESET
      |                                                                                        ^
In file included from ../../../../../src/mame/misc/blitz68k.cpp:1:
In file included from /Users/wyattwong/GitHub/mame/build/projects/sdl/mame/gmake-osx-clang/../../../../../src/emu/emu.h:58:
/Users/wyattwong/GitHub/mame/build/projects/sdl/mame/gmake-osx-clang/../../../../../src/emu/ioport.h:1202:13: note: expanded from macro 'PORT_RESET'
 1202 |         configurer.field_set_analog_reset();
      |                    ^
/Users/wyattwong/GitHub/mame/build/projects/sdl/mame/gmake-osx-clang/../../../../../src/emu/ioport.h:1059:4: note: 'field_set_analog_reset' has been explicitly marked deprecated here
 1059 |         [[deprecated("PORT_RESET is deprecated; manage counter state explicitly")]]
      |           ^

**Comment by rb6502** [comment]
Those are all MAME-asserted deprecations, they're not related to any specific compiler and function mostly as reminders.

**Reply by wyatt-wong** [reply]
> Those are all MAME-asserted deprecations, they're not related to any specific compiler and function mostly as reminders.

Sorry what does that mean EXACTLY ? You mean any C/C++ compiler that report such deprecated warnings will continue to report forever but they will still compile MAME into binary successfully ?

**Reply by cuavas** [reply]
It’s to remind people not to use those functions in new code and consider updating legacy code to remove them.

**Reply by wyatt-wong** [reply]
> It’s to remind people not to use those functions in new code and consider updating legacy code to remove them.

But those who participate to enhance MAME allow those deprecated code to exists which totally contradicts with what you said.

**Reply by cuavas** [reply]
It takes time to remove them all – MAME is big.
